examples/broadcasting_your_own_methods.py

```diff
@@ -0,0 +1,472 @@
+)
+first_entry = individual_0_centroid["time"][individual_0_centroid].values[0]
+last_exit = individual_0_centroid["time"][individual_0_centroid].values[-1]
+print(
+    f"{i_id} first entered the slippery region at "
+    f"{first_entry} and last exited at {last_exit}"
+)
+
+# %%
+# Data Generalisation Issues
+# --------------------------
+# The shape of the resulting ``DataArray`` is the same as our original
+# ``DataArray``, but without the ``"space"`` dimension.
+# Indeed, we have essentially collapsed the ``"space"`` dimension, since our
+# ``in_slippery_region`` function takes in a 1D data slice (the x, y positions
+# of a single individual's centroid at a given point in time) and returns a
+# scalar value (True/False).
+# However, the fact that we have to construct a new ``DataArray`` after running
+# our function over all space slices in our ``DataArray`` is not scalable - our
+# ``for`` loop approach relied on knowing how many dimensions our data had (and
+# the size of those dimensions). We don't have a guarantee that the next
+# ``DataArray`` that comes in will have the same structure.
+
+# %%
+# Making our Function Broadcastable
+# ---------------------------------
+# To combat this problem, we can make the observation that given any
+# ``DataArray``, we always want to broadcast our ``in_slippery_region``
+# function
+# along the ``"space"`` dimension. By "broadcast", we mean that we always want
+# to run our function for each 1D-slice in the ``"space"`` dimension, since
+# these are the (x, y) coordinates. As such, we can decorate our function with
+# the ``make_broadcastable`` decorator:
+
+
+@make_broadcastable()
+def in_slippery_region_broadcastable(xy_position) -> float:
+    return in_slippery_region(xy_position=xy_position)
+
+
+# %%
+# Note that when writing your own methods, there is no need to have both
+# ``in_slippery_region`` and ``in_slippery_region_broadcastable``, simply apply
+# the ``make_broadcastable`` decorator to ``in_slippery_region`` directly.
+# We've made two separate functions here to illustrate what's going on.
+
+# %%
+# ``in_slippery_region_broadcastable`` is usable in exactly the same ways as
+# ``in_slippery_region`` was:
+
+for point in [(0, 100), (450, 700), (550, 1500), (601, 500)]:
+    print(
+        f"{point} is in slippery region: "
+        f"{in_slippery_region_broadcastable(point)}"
+    )
+
+
+# %%
+# However, ``in_slippery_region_broadcastable`` also takes a ``DataArray`` as
+# the first (``xy_position``) argument, and an extra keyword argument
+# "``broadcast_dimension``". These arguments let us broadcast across the given
+# dimension of the input ``DataArray``, treating each 1D-slice as a separate
+# input to ``in_slippery_region``.
+
+in_slippery_region_broadcasting = in_slippery_region_broadcastable(
+    positions,  # Now a DataArray input
+    broadcast_dimension="space",
+)
+
+print("DataArray output using broadcasting: ")
+print(in_slippery_region_broadcasting)
```

Comment:
To get an interactive html widget
```suggestion
in_slippery_region_broadcasting
```